agent-core/src/main/java/com/newrelic/agent/android/harvest/HarvestTimer.java

```diff
@@ -145,20 +140,46 @@ public void shutdown() {
         scheduler.shutdownNow();
     }
 
-    // Runs a tick of the Harvester immediately, disregarding any 'time since last tick' limits.
-    public void tickNow() {
-        final HarvestTimer timer = this;
-        ScheduledFuture future = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
-                timer.tick();
-            }
-        }, 0, TimeUnit.SECONDS);
+    public void restart() {
```

Comment:
This new method encapsulated changes to HarvestTimer to _restart the session_, which is to say resetting the the harvest times. Session times (total, time since start) are derived from this clock. It has not been hooked up to exiting code.